### DISCUSSION TITLE
Applications spanning multiple columns

### DISCUSSION THREAD

**Discussion by tmao8** [AMDResearch/Riallto#32] [Q&A] [answered]
Is it possible to develop applications spanning multiple columns? Using tloc I am restricted to keeping kernels in one column (for example, when setting the kernel location to (1, 4) I get the error "ValueError: Cannot place kernel example_0 at (1, 4) - (1, 4) does not exist")

**Comment by STFleming** [comment, marked as answer]
Hi @tmao8, 

Currently this is not permitted in the latest release of Riallto and applications are restricted to a single column. This is something that could potentially be explored in future releases. 

All the best,
Shane

**Reply by soumitraghosh388** [reply]
I can see latest version 1.1 released in July 2024. Is it still not supported?

**Reply by mariodruiz** [reply]
@soumitraghosh388, no multi column is not supported in Riallto

**Comment by hecmay** [comment]
Hi @STFleming, 

Can you please explain the reason that an application is restricted to single column in Riallto? Is there any fix to bypass this restriction? I am not so sure why since Riallto is built on top of MLIR-AIE, which does not have any restrictions like this.

I also notice that you are pushing some major changes to Riallto codebase lately. When can we expect this to be resolved?

Thanks,
Shawn

**Reply by mariodruiz** [reply]
Hi @hecmay,
We have not tested neither verified multi column. We are also using a relative old version of MLIR-AIE so, I am not sure if multi column is supported. Support for multi column is a great contribution in case you have time.
